src/App.tsx

```diff
@@ -1,34 +1,71 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
-import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [search, setSearch] = useState('');
+  const [filter, setFilter] = useState('all');
+
+  const loaderData = (filterText: string) => {
+    switch (filterText) {
+      case 'completed':
+        getTodos().then(res => {
```

Comment:
You shouldn't make query to back end every time you change filter. Instead you can filter all todos you've queried for the first time.